backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -60,16 +60,24 @@ def get_artifact_from_job_as_content(
         ).encode()
 
     def __get_file_from_repository(
-        self, project_id: str, trusted_file_path: str, headers: dict = None
+        self,
+        project_id: str,
+        trusted_file_path: str,
+        revision: str | None = None,
+        headers: dict | None = None,
     ) -> requests.Response:
+        branch = revision if revision else self.git_model.revision
```

Comment:
```suggestion
        revision = revision or self.git_model.revision
```

Ever better would it be to put `revision or self.git_model.revision` directly in the requests call.